solidity/contracts/converter/BancorConverter.sol

```diff
@@ -590,11 +606,13 @@ contract BancorConverter is IBancorConverter, SmartTokenController, ContractRegi
       * @param _reserveToken    reserve token contract address
       * @param _sellAmount      amount of tokens to sell (in units of the smart token)
       * @param _minReturn       if the conversion results in an amount smaller the minimum return - it is cancelled, must be nonzero
+      * @param _beneficiary     wallet to receive the conversion result
       * 
       * @return amount of tokens received (in units of the reserve token)
     */
-    function sell(IERC20Token _reserveToken, uint256 _sellAmount, uint256 _minReturn) internal returns (uint256) {
-        require(_sellAmount <= token.balanceOf(msg.sender)); // validate input
+    function sell(IERC20Token _reserveToken, uint256 _sellAmount, uint256 _minReturn, address _beneficiary) internal returns (uint256) {
+        // ensure that the input amount was already deposited
+        require(_sellAmount <= token.balanceOf(this));

```

Comment:
@yudilevi:
Why have you changed `msg.sender` to `this` here?

Reply:
I thought it would be more consistent that the converter handle the smart token the same way it handles other tokens.
Plus it's more "correct" in a way that the converter destroys its own holdings instead of destroying the tokens held by another account.
We could in the future remove the option for smart tokens to get destroyed by the converter entirely and only allow the holder of smart tokens to destroy their own tokens.

Reply:
@yudilevi:
But here `token.balanceOf(msg.sender)` and `token.balanceOf(this)` are two different things.

Reply:
because we don't destroy the sender balance anymore - we destroy self balance or fail if it's not already deposited